dagster_utils/resources/beam.py

```diff
@@ -0,0 +1,203 @@
+            'autoscalingAlgorithm': 'THROUGHPUT_BASED',
+            'numWorkers': str(self.cloud_config.starting_workers),
+            'maxNumWorkers': str(self.cloud_config.max_workers),
+            'experiments': 'shuffle_mode=service',
+        }
+
+        # process the args dict into dataflow flags
+        args = [
+            f'--{field_name}={value}'
+            for field_name, value
+            in args_dict.items()
+        ]
+
+        image_name = f"{self.image_name}:{self.image_version}"  # {context.solid_config['version']}"
+        job = self.dispatch_k8s_job(image_name, job_name, args)
+        self.logger.info("Dataflow job started")
+
+        client = DagsterKubernetesClient.production_client()
+        client.wait_for_job_success(job.metadata.name, self.namespace)
+
+    def dispatch_k8s_job(self, image_name: str, job_name_prefix: str, args: List[str]) -> V1Job:
+        # we will need to poll the pod/job status on creation
+        kubernetes.config.load_kube_config()
+
+        job_name = f"{job_name_prefix}-{uuid4()}"
+        pod_name = f"{job_name}-pod"
+        job_container = kubernetes.client.V1Container(
+            name=job_name,
+            image=image_name,
+            args=args,
+        )
+
+        # TODO unhardcode the SA name below
+        template = kubernetes.client.V1PodTemplateSpec(
+            metadata=kubernetes.client.V1ObjectMeta(name=pod_name),
+            spec=kubernetes.client.V1PodSpec(
+                restart_policy="Never",
+                containers=[job_container],
+                service_account_name=self.kubernetes_service_account,
+            ),
+        )
+
+        job = kubernetes.client.V1Job(
+            api_version="batch/v1",
+            kind="Job",
+            metadata=kubernetes.client.V1ObjectMeta(
+                name=job_name,
+            ),
+            spec=kubernetes.client.V1JobSpec(
+                template=template,
+                backoff_limit=2,
+                ttl_seconds_after_finished=86400,  # one day
+            ),
+        )
+        batch_v1 = kubernetes.client.BatchV1Api()
+        api_response = batch_v1.create_namespaced_job(
+            body=job,
+            namespace=self.namespace,
+        )
+        self.logger.info(f"Job created. status='{str(api_response.status)}'")
+
+        return api_response
+
+
+@resource({
+    "project": Field(StringSource),
+    "subnet_name": Field(StringSource),
+    "region": Field(StringSource),
+    "worker_machine_type": Field(StringSource),
+    "starting_workers": Field(IntSource),
+    "max_workers": Field(IntSource),
+    "service_account": Field(StringSource),
+    "kubernetes_service_account": Field(StringSource),
+    "temp_bucket": Field(StringSource),
+    "image_name": Field(StringSource),
+    "image_version": Field(StringSource),
+    "namespace": Field(StringSource),
+})
+def dataflow_beam_runner(init_context: InitResourceContext) -> DataflowBeamRunner:
```

Comment:
Nice! I can pull this into the DAP project's dagster implementation and extend it to create the local_beam_runner there, right?

Reply:
local beam runner should be later in the file - you can just pull in that resource directly once this dependency is published and use it in a mode in the DAP dagster application